### PR TITLE
Monitor Edison battery WITHOUT cron

### PR DESCRIPTION
**Pull request by baughaw**

### CODE REVIEW

**Comment by scottleibrand** [comment]
Are there any situations where the aliases we're replacing are different (perhaps for MDT CGM or something)?

**Comment by baughaw** [comment]
Not that I could see.  The two aliases seem independent of sensor choice.

**Comment by baughaw** [comment]
It is possible..  let me check :(

**Comment by baughaw** [comment]
Yep it is missing.  

openaps report add monitor/edison-battery2.json JSON edison-battery shell json batteryVoltage battery

**Comment by scottleibrand** [comment]
K. Can you commit that change to your branch so it gets added to the PR?

File: bin/oref0-setup.sh [hunk 1]
```diff
@@ -356,6 +356,35 @@ elif [[ ${CGM,,} =~ "G4" || ${CGM,,} =~ "shareble" ]]; then
...
+      cd ~/src/EdisonVoltage
+      make voltage
+   fi
+ 
+    
+   echo Adding Edison battery device
+   openaps device add edison-battery process sudo $HOME/EdisonVoltage/voltage || die "Can't add edison battery"
+   
+   echo Adding Edison battery report
+   openaps report add monitor/edison-battery2.json JSON edison-battery shell json batteryVoltage battery
```
Comment (scottleibrand) [inline bin/oref0-setup.sh]: I think you need to remove the 2?

**Comment by baughaw** [comment]
I believe I committed it..  Did I do something wrong?

**Comment by scottleibrand** [comment]
Just had only seen your comment not your commit when I posted that. Flaky LTE.

File: bin/oref0-setup.sh (hunk 1 above)
Comment (baughaw) [inline bin/oref0-setup.sh]: ha, yep, I was testing on my live rig. :)   Hang on

**Comment by Bender1061** [comment]
how about to keep it out of a cron job, not have edison-battery as part of pump-loop alias or better yet ns-loop alias. That way it keeps it out of monitor-pump, which the Battery is not part of, and semi keeps it separate from the pump just like the BT and the wifi are?

**Comment by Bender1061** [comment]
should probably also make the changes to load the changes from a Json file, like how we add all the other reports and alias from. that way it keeps the oref0-setup script as clean as we can (granted that is getting difficult with the amount of awesomeness everyone had been adding to it)

**Comment by scottleibrand** [comment]
Yeah, I think @Bender1061's suggestion of putting edison-battery in the ns-loop alias, instead of in monitor-pump, makes more sense.  @baughaw you want to adjust that, or should I?

**Comment by Bender1061** [comment]
add an echo to the edison-battery, so that you can see it posting to the ns-loop.log 

like this

echo Edison Battery at $(date): $(sudo ~/src/EdisonVoltage/voltage json batteryVoltage battery) && sudo ~/src/EdisonVoltage/voltage json batteryVoltage battery

leave the sudo in there, so it will work with people not running the explorer board.

**Comment by baughaw** [comment]
@scottleibrand Hi Scott, I am traveling today..  So you can make the change if you like or wait for me until tomorrow.  I wont be offended :)

**Comment by baughaw** [comment]
@Bender1061 Would this cause the code to run twice?

echo Edison Battery at $(date): $(sudo ~/src/EdisonVoltage/voltage json batteryVoltage battery) && sudo ~/src/EdisonVoltage/voltage json batteryVoltage battery

Seems like that is an expensive operation since the voltage code writes and reads from 2 files each time it runs.

**Comment by scottleibrand** [comment]
Firing up new python environments for each `openaps` call is probably a lot more expensive than reading voltage twice. :-) That said, running it once and then doing a `cat` of the output file would ensure that what is in the log matches what is reported to NS.

**Pull request closed without merging**